### PR TITLE
Digital Camo Refactor

### PR DESCRIPTION
**Pull request by PowerfulBacon**
Ports a PR that ports

tgstation/tgstation#46134
tgstation/tgstation#46328
tgstation/tgstation#46814

but then optimises it so that the new code isn't worse than the old code.

## About The Pull Request

Refactors digital camo to not use Life().
Mostly a port of a kapu PR which is a TG PR, except the process() is removed which made the optimisation twice as slow.

## Why It's Good For The Game

Optimisation that's actually faster unlike the TG version.

## Testing Photographs and Procedure

Test enabled:

image

Test Disabled:

image

Test AI:

image

Test multicam:

image

Test normal again:

image

Test dropped again:

image

Test HUD:

image

image

(Fixed the hud, port missed a part from TG.)

image

## Changelog
:cl: PowerfulBacon, Kapu, nemvar, AnturK
refactor: Refactors digital invisibility to be faster
refactor: All silicons are now affected by digital camoflague including Cyborgs and pAIs.
/:cl:

### CODE REVIEW

**Comment by Kapu1178** [comment]
Undocumented change: Cyborgs now are vulnerable to digital camo.

File: code/datums/elements/digital_camo.dm [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+/datum/element/digital_camo/proc/UnhideFromSiliconHuds(mob/living/target)
+	for(var/mob/living/silicon/silicon as() in GLOB.silicon_mobs)
+		var/datum/atom_hud/M = GLOB.huds[silicon.med_hud]
+		var/datum/atom_hud/S = GLOB.huds[silicon.sec_hud]
+		M.unhide_single_atomhud_from(silicon, target)
+		S.unhide_single_atomhud_from(silicon, target)
+
+/datum/element/digital_camo/proc/on_examine(datum/source, mob/user, list/examine_list)
+	SIGNAL_HANDLER
+	examine_list += "<span class='warning'>[source.p_their()] skin seems to be shifting and morphing like is moving around below it.</span>"
```
Comment (Sarchutar) [inline code/datums/elements/digital_camo.dm:67]: This sentence sounds wrong/like something is missing

Comment (PowerfulBacon) [reply code/datums/elements/digital_camo.dm:67]: It's a direct copy and paste from before

Comment (utilbird) [reply code/datums/elements/digital_camo.dm:67]: yeah it should be `like it is`.

Even then, that sounds like machine translated text

File: code/datums/hud.dm [hunk 2]
```diff
@@ -56,6 +57,21 @@ GLOBAL_LIST_INIT(huds, list(
 	GLOB.all_huds -= src
 	return ..()
 
+/datum/atom_hud/proc/hide_single_atomhud_from(hud_user,hidden_atom)
```
Comment (Archanial) [inline code/datums/hud.dm]:
```suggestion
/datum/atom_hud/proc/hide_single_atomhud_from(hud_user, hidden_atom)
```

**Review by Archanial — DISMISSED** [review]

**Review by Archanial — APPROVED** [review]

**Pull request merged**